mantidproject/mantid
# Enable mixing of automatic and user bin widths in DirectILLReduction

This PR adds a new property to `DirectILLReduction` which allows users to mix automatic binning with their own settings. Generally, the automatic binning works fine but sometimes there is need for more fine-grained binning around the elastic line.

**Report to:** Björn Fåk/ILL.

**To test:**

The test data used below can be found in the unit test data. The syntax for the `EnergyBinning` property is a mixture of rebinning params and the letter `a` which can be thought as a an automatic bin width. Play around with the parameters in the script below and see if the results make sense.

```python
ws = DirectILLCollectData('ILL/IN4/084447.nxs')
sqw = DirectILLReduction(ws, EnergyRebinning='20, -10, a, 4, 0.5, 8')
```

Fixes #23560.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
